File: exercises-cpp/3. Repetitivas/10.cpp
#include<iostream>
using namespace std;

int main() {
    system("cls");
    int a, b, c, m, contador=0;

    for (int i=1000; i<=10000; i++){
        c = i%10;
        b = (i%100-c)/10;
        a = (i%1000-b*10-c)/100;
        m= i/1000;

        if ( m==1 || m==3 || m==5 || m==7 || m==9 ){
            if (a%2==0){
                if (b%2!=0 && c%2==0){
                    if (a+c == m+b) { cout<<i<<"; "; contador++;}
                }else if (b%2==0 && c%2!=0) {
                    if (a+b == m+c) { cout<<i<<"; "; contador++;}
                }
            }else if (a%2!=0){
                if (b%2==0 && c%2==0){
                    if ( b+c == m+a) { cout<<i<<"; "; contador++;}
                }
            }
        }else if ( m==2 || m==4 || m==6 || m==8 ){
            if (a%2==0){
                if (b%2!=0 && c%2!=0){
                    if ( m+a == b+c ) { cout<<i<<"; "; contador++;}
                }
            }else if (a%2!=0){
                if (b%2!=0 && c%2==0){
                    if ( m+c == a+b) {cout<<i<<"; "; contador++;}
                }else if (b%2==0 && c%2!=0) {
                    if (m+b == a+c) {cout<<i<<"; "; contador++;}
                }
            }
        }
    }

    cout<<"\nNumeros que cumplen la condicion: "<<contador<<endl;
    return 0;
}

File: exercises-python/3. Repetitivos/10.py
'''
10.Desarrolle el programa que muestre todos los números de 4 cifras
que cumplen la condición de que la suma de las cifras pares es igual
a la suma de las cifras impares, muestre además la cantidad de números encontrados.
'''
import os
os.system("cls")

par=0
impar=0
count = 0

for i in range(1000,10000):
    i=str(i)
    for x in i:
        a = int(x)
        if a % 2 == 0:
            par += a
        else:
            impar += a
    if par == impar:
        print(i,end=" ")
        count += 1
        if count % 20 == 0: print()
    par = 0
    impar = 0
print()
print()
print(f"Se encontraron {count} números.")